src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double MULTIPLIER_FOR_ARRAY_SIZE = 1.5;
+    private int arraySize = 10;
+    private T[] array;
+    private int numberOfElements;
+
```

Comment:
I'd rather rename numberOfElements to **size**, and arraySize to **arrayCapacity**, it provides more readability, but it also may be a matter of taste.